internal/services/verifible_credential_service.go

```diff
@@ -0,0 +1,194 @@
+package services
+
+//go:generate mockgen -source verifible_credential_service.go -destination mocks/verifible_credential_service_mock.go
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"crypto/rand"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	ethC "github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/crypto/secp256k1"
+	"github.com/mr-tron/base58"
```

Comment:
I would use [the Bitcoin library](https://pkg.go.dev/github.com/btcsuite/btcutil/base58).